test/SymbolGtest.cpp

```diff
@@ -0,0 +1,184 @@
+        static constexpr int FOURTH_INT_VALUE = 44;
+        static constexpr double FIRST_DOUBLE_VALUE = 42.42e10;
+        static constexpr double SECOND_DOUBLE_VALUE = 43.43e10;
+        static constexpr double THIRD_DOUBLE_VALUE = 41.41e10;
+        static constexpr double FOURTH_DOUBLE_VALUE = 44.44e10;
+        static constexpr const char *BAD_USER_INPUT = "asdf";
+
+        calculator::Symbol<int> intSymbolA;
+        calculator::Symbol<int> intSymbolBBound;
+        calculator::Symbol<int> intSymbolDBound;
+        calculator::Symbol<double> doubleSymbolA;
+        calculator::Symbol<double> doubleSymbolDBound;
+        calculator::Symbol<double> doubleSymbolBBound;
+
+        // Using SetUp() is normally preferred to having a ctor.
+        // In this case, we can't get away from initializing in the ctor since 
+        // the instance variables require symbols to be set on construction
+        GivenInitializedSymbolsSomeWithValues(): 
+                intSymbolA{'a'},
+                intSymbolBBound{'B'},
+                intSymbolDBound{'d'},
+                doubleSymbolA{'A'},
+                doubleSymbolBBound{'B'},
+                doubleSymbolDBound{'d'}
+        {
+        }
+
+        void SetUp() override
+        {
+            intSymbolBBound.setValue(FIRST_INT_VALUE);
+            intSymbolDBound.setValue(FIRST_INT_VALUE);
+            doubleSymbolBBound.setValue(FIRST_DOUBLE_VALUE);
+            doubleSymbolDBound.setValue(FIRST_DOUBLE_VALUE);
+        }
+};
+constexpr int GivenInitializedSymbolsSomeWithValues::FIRST_INT_VALUE;
+constexpr int GivenInitializedSymbolsSomeWithValues::SECOND_INT_VALUE;
+constexpr int GivenInitializedSymbolsSomeWithValues::THIRD_INT_VALUE;
+constexpr int GivenInitializedSymbolsSomeWithValues::FOURTH_INT_VALUE;
+constexpr double GivenInitializedSymbolsSomeWithValues::FIRST_DOUBLE_VALUE;
+constexpr double GivenInitializedSymbolsSomeWithValues::SECOND_DOUBLE_VALUE;
+constexpr double GivenInitializedSymbolsSomeWithValues::THIRD_DOUBLE_VALUE;
+constexpr double GivenInitializedSymbolsSomeWithValues::FOURTH_DOUBLE_VALUE;
+constexpr const char *GivenInitializedSymbolsSomeWithValues::BAD_USER_INPUT;
+
+
+
+TEST_F(GivenInitializedSymbolsSomeWithValues,
+        WhenHasValueIsCalledWithoutSettingValue_ThenFalseIsReturned)
+{
+    EXPECT_FALSE(intSymbolA.hasValue());
+    EXPECT_FALSE(doubleSymbolA.hasValue());
+}
+
+
+
+TEST_F(GivenInitializedSymbolsSomeWithValues,
+        WhenHasValueIsCalledAfterSettingValue_ThenTrueIsReturned)
+{
+    EXPECT_TRUE(intSymbolBBound.hasValue());
+    EXPECT_TRUE(doubleSymbolBBound.hasValue());
+}
+
+
+
+TEST_F(GivenInitializedSymbolsSomeWithValues,
+        WhenSetValueIsCalledForTheFirstTime_ThenInternalValueIsUpdatedAndNoErrorsHappen)
+{
+    calculator::Symbol<int> intSymbolC{'C'};
+    calculator::Symbol<double> doubleSymbolC{'C'};
+    intSymbolC.setValue(FIRST_INT_VALUE);
+    doubleSymbolC.setValue(FIRST_DOUBLE_VALUE);
+}
```

Comment:
I don't see an expect here, what should the outcome be for it to be checked

Reply:
Hmm, I would need to .getValue() and use EXPECT_EQ to check that the value was set correctly, but then the test would include both getValue() and setValue() method calls. I could either keep them broken up, like it is now, or use both getValue() and setValue() in the same test case. What do you think I should do here?